### PR TITLE
Implement clear() for LCD 'LiquidCrystalClear'

### PR DESCRIPTION
**Pull request by amishv**
## Description:
Implementation of clear() in LCD display.
https://www.arduino.cc/en/Reference/LiquidCrystalClear
**Related issue (if applicable):** fixes <link to issue>
Character-Based LCD Display PCF8574 - Clear command not found #712
**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/lcd_base/lcd_display.cpp [hunk 1]
```diff
@@ -156,6 +156,11 @@ void LCDDisplay::strftime(uint8_t column, uint8_t row, const char *format, time:
     this->print(column, row, buffer);
 }
 void LCDDisplay::strftime(const char *format, time::ESPTime time) { this->strftime(0, 0, format, time); }
+void LCDDisplay::clear(void){
```
Comment (OttoWinter) [inline esphome/components/lcd_base/lcd_display.cpp]: This function is not defined in the base class: https://github.com/esphome/esphome/blob/fb0f0ee785388bf0d6060556b1c912d9e51eb299/esphome/components/lcd_base/lcd_display.h has no clear method.

Comment (amishv) [reply esphome/components/lcd_base/lcd_display.cpp]: Thanks,
 I have implemented that, not sure how it compiled locally. 
A C kernel programmer gets confused by C++

Comment (amishv) [reply esphome/components/lcd_base/lcd_display.cpp]: I am unable to find how to point to a certain repository to test the code. There was an option in esphomelib but unable to find in esphome

Comment (OttoWinter) [reply esphome/components/lcd_base/lcd_display.cpp]: See the contributing page - you can use the instructions there to install a development setup or you can use the `custom_components` folder to override an integration's code without having to install a dev setup (also see contributing guide)

**Comment by amishv** [comment]
Caught in middle of multiple changes. Will retry the checkin

**Pull request closed without merging**